main.c

```diff
@@ -170,10 +170,23 @@ void fetch_one_suite(const char * suite_path, struct application_context * appli
 
 void read_suites(struct application_context * application_context)
 {
+    /* replacing a standard array with something more appropriate like a hashmap is probably what we need but this should do for now. */
+    const char * fnames[1024];
+    int size = 0;
```

Comment:
It would be better to understand if rename it to `filenames_cache_pos` or `filenames_cache_cursor` for example.